From 009e79454070147555ab856a2d56046f357554a0 Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
Date: Wed, 6 Feb 2019 15:58:54 +0100
Subject: [PATCH 096/188] dt: mba6ulx: add display trees for all variants

for imx6ul-mba6ulx, imx6ul-mbaulx-lga, imx6ull-mba6ulx, imx6ull-mba6ulx-lga
each with parallel displays dmb-ct21, dmb-ct44, glyn-etm0700 and lvds.

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
---
 arch/arm/boot/dts/Makefile                    | 18 +++-
 arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct21.dtsi | 64 +++++++++++++
 arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct44.dtsi | 66 ++++++++++++++
 .../dts/imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi | 64 +++++++++++++
 arch/arm/boot/dts/imx6ul-mba6ul-lcdif.dtsi    | 90 +++++++++++++++++++
 arch/arm/boot/dts/imx6ul-mba6ul-lvds.dtsi     | 60 +++++++++++++
 arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct21.dts | 14 +++
 arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct44.dts | 14 +++
 .../dts/imx6ul-mba6ulx-glyn-etm0700g0edh6.dts | 14 +++
 .../boot/dts/imx6ul-mba6ulx-lga-dmb-ct21.dts  | 14 +++
 .../boot/dts/imx6ul-mba6ulx-lga-dmb-ct44.dts  | 14 +++
 .../imx6ul-mba6ulx-lga-glyn-etm0700g0edh6.dts | 15 ++++
 arch/arm/boot/dts/imx6ul-mba6ulx-lga-lvds.dts | 14 +++
 arch/arm/boot/dts/imx6ul-mba6ulx-lvds.dts     | 14 +++
 .../arm/boot/dts/imx6ull-mba6ulx-dmb-ct21.dts | 14 +++
 .../arm/boot/dts/imx6ull-mba6ulx-dmb-ct44.dts | 14 +++
 .../imx6ull-mba6ulx-glyn-etm0700g0edh6.dts    | 14 +++
 .../boot/dts/imx6ull-mba6ulx-lga-dmb-ct21.dts | 14 +++
 .../boot/dts/imx6ull-mba6ulx-lga-dmb-ct44.dts | 14 +++
 ...imx6ull-mba6ulx-lga-glyn-etm0700g0edh6.dts | 15 ++++
 .../arm/boot/dts/imx6ull-mba6ulx-lga-lvds.dts | 14 +++
 arch/arm/boot/dts/imx6ull-mba6ulx-lvds.dts    | 14 +++
 22 files changed, 587 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct21.dtsi
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct44.dtsi
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ul-lcdif.dtsi
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ul-lvds.dtsi
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct21.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct44.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-glyn-etm0700g0edh6.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct21.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct44.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-lga-glyn-etm0700g0edh6.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-lga-lvds.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-mba6ulx-lvds.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct21.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct44.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-glyn-etm0700g0edh6.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct21.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct44.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-lga-glyn-etm0700g0edh6.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-lga-lvds.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-mba6ulx-lvds.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index a847a8e36f9a..069e379ec78e 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -573,7 +573,15 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-9x9-evk-btwifi-oob.dtb \
 	imx6ul-9x9-evk-ldo.dtb \
 	imx6ul-mba6ulx-lga.dtb \
-	imx6ul-mba6ulx.dtb
+	imx6ul-mba6ulx-lga-dmb-ct21.dtb \
+	imx6ul-mba6ulx-lga-dmb-ct44.dtb \
+	imx6ul-mba6ulx-lga-glyn-etm0700g0edh6.dtb \
+	imx6ul-mba6ulx-lga-lvds.dtb \
+	imx6ul-mba6ulx.dtb \
+	imx6ul-mba6ulx-dmb-ct21.dtb \
+	imx6ul-mba6ulx-dmb-ct44.dtb \
+	imx6ul-mba6ulx-glyn-etm0700g0edh6.dtb \
+	imx6ul-mba6ulx-lvds.dtb
 dtb-$(CONFIG_SOC_IMX6ULL) += \
 	imx6ull-14x14-ddr3-arm2.dtb \
 	imx6ull-14x14-ddr3-arm2-adc.dtb \
@@ -602,7 +610,15 @@ dtb-$(CONFIG_SOC_IMX6ULL) += \
 	imx6ull-9x9-evk-btwifi-oob.dtb \
 	imx6ull-9x9-evk-ldo.dtb		\
 	imx6ull-mba6ulx.dtb \
+	imx6ull-mba6ulx-dmb-ct21.dtb \
+	imx6ull-mba6ulx-dmb-ct44.dtb \
+	imx6ull-mba6ulx-glyn-etm0700g0edh6.dtb \
+	imx6ull-mba6ulx-lvds.dtb \
 	imx6ull-mba6ulx-lga.dtb \
+	imx6ull-mba6ulx-lga-dmb-ct21.dtb \
+	imx6ull-mba6ulx-lga-dmb-ct44.dtb \
+	imx6ull-mba6ulx-lga-glyn-etm0700g0edh6.dtb \
+	imx6ull-mba6ulx-lga-lvds.dtb \
 	imx6ulz-14x14-evk.dtb \
 	imx6ulz-14x14-evk-btwifi.dtb \
 	imx6ulz-14x14-evk-btwifi-oob.dtb \
diff --git a/arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct21.dtsi b/arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct21.dtsi
new file mode 100644
index 000000000000..bf25870c1941
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct21.dtsi
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ * Copyright 2017 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	/* S070WV106H-CT21 */
+	display_ct21: display {
+		bits-per-pixel = <32>; /* 32: RGB888/666 mode */
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing1>;
+			/* S070WS12H-O-CT21 rev. 2.0 2016/02/18
+			 * S070PWS19HP-FC21 ver 2.0 2017/2/21
+			 * Truecolor FB needed - pin config on display adapter
+			 */
+			timing1: 1024x600p62 {
+				clock-frequency = <45000000 51200000 57000000>;
+				hactive = <1024>;
+				vactive = <600>;
+				hfront-porch = <140 160 180>;
+				hsync-len = <20>;
+				hback-porch = <140>;
+				vfront-porch = <2 12 22>;
+				vsync-len = <3>;
+				vback-porch = <20>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
+&backlight {
+	/* 50000 ... 1000 nsec -> 20 ... 1000 kHz */
+	pwms = <&pwm2 0 20000>;
+	brightness-levels = <0 2 4 8 16 32 64 128 255>;
+	default-brightness-level = <8>;
+	status = "okay";
+};
+
+&lcdif {
+	display = <&display_ct21>;
+	status = "okay";
+};
+
+&polytouch {
+	wake-gpios = <&expander_out0 1 GPIO_ACTIVE_HIGH>;
+	/* reset-gpios is controlled by stk reset */
+	gain = <20>;
+
+	touchscreen-size-x = <1024>;
+	touchscreen-size-y = <600>;
+
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct44.dtsi b/arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct44.dtsi
new file mode 100644
index 000000000000..eab2fb1ccbbe
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ul-dmb-ct44.dtsi
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	/* compatible:
+	 * S070WV106H-CT44 v1.0 2017/02/21
+	 * S070SWV29HG-DC44 v 3.0 2017/02/21
+	 * Truecolor FB needed - pin config on display adapter
+	 */
+	display_ct44: display {
+		bits-per-pixel = <32>; /* 32: RGB888/666 mode */
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing1>;
+			timing1: 800x480p62 {
+				clock-frequency = <26300000 33300000 46800000>;
+				hactive = <800>;
+				vactive = <480>;
+				hfront-porch = <210>;
+				hsync-len = <2>;
+				hback-porch = <44>;
+				vfront-porch = <22>;
+				vsync-len = <2>;
+				vback-porch = <21>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
+&backlight {
+	/* 50000 ... 1000 nsec -> 20 ... 1000 kHz */
+	pwms = <&pwm2 0 20000>;
+	brightness-levels = <0 1 4 8 16 32 64 128 255>;
+	default-brightness-level = <8>;
+	status = "okay";
+};
+
+&lcdif {
+	display = <&display_ct44>;
+	status = "okay";
+};
+
+&polytouch {
+	wake-gpios = <&expander_out0 1 GPIO_ACTIVE_HIGH>;
+	/* reset-gpios is controlled by stk reset */
+	gain = <20>;
+
+	touchscreen-size-x = <800>;
+	touchscreen-size-y = <480>;
+
+	status = "okay";
+};
+
diff --git a/arch/arm/boot/dts/imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi b/arch/arm/boot/dts/imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi
new file mode 100644
index 000000000000..e6e62f9aee63
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	/* ETM0700G0EDH6, DS 2016/04/06, ver 3 */
+	display_etm0700: display {
+		bits-per-pixel = <32>; /* 32: RGB8880 */
+		bus-width = <24>; 
+
+		display-timings {
+			native-mode = <&timing1>;
+			timing1: 800x480p62 {
+				clock-frequency = <33260000>;
+				hactive = <800>;
+				vactive = <480>;
+				hfront-porch = <40>;
+				hsync-len = <128>;
+				hback-porch = <88>;
+				vfront-porch = <10>;
+				vsync-len = <2>;
+				vback-porch = <33>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
+&backlight {
+	/* 1 kHz as stated by GLYN FAE */
+	pwms = <&pwm2 0 1000000>;
+	brightness-levels = <255 200 180 160 128 96 64 32 16>;
+	default-brightness-level = <5>;
+	status = "okay";
+};
+
+&lcdif {
+	display = <&display_etm0700>;
+	status = "okay";
+};
+
+&polytouch {
+	/* reset-gpios is controlled by stk reset */
+	gain = <1>;
+	offset = <0>;
+	threshold = <0x0f>;
+
+	/*
+	 * do not use size settings, display touch reads
+	 * out and uses 1792 x 1024 (tested with ETM0700G0EDH6)
+	 */
+
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6ul-mba6ul-lcdif.dtsi b/arch/arm/boot/dts/imx6ul-mba6ul-lcdif.dtsi
new file mode 100644
index 000000000000..9315b8ffc973
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ul-lcdif.dtsi
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ * Copyright 2017 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	backlight: pwm_backlight {
+		compatible = "pwm-backlight";
+		power-supply = <&reg_mba6ul_3v3>;
+		enable-gpios = <&expander_out0 4 GPIO_ACTIVE_HIGH>;
+		status = "disabled";
+	};
+
+	pxp_v4l2 {
+		compatible = "fsl,imx6ul-pxp-v4l2", "fsl,imx6sx-pxp-v4l2", "fsl,imx6sl-pxp-v4l2";
+		status = "okay";
+	};
+};
+
+&analog_touch {
+	status = "disabled";
+};
+
+&i2c4 {
+	polytouch: edt-ft5x04@38 {
+		compatible = "edt,edt-ft5406", "edt,edt-ft5x06";
+		reg = <0x38>;
+		interrupt-parent = <&expander_in0>;
+		interrupts = <3 IRQ_TYPE_EDGE_FALLING>;
+		status = "disabled";
+	};
+};
+
+&iomuxc {
+	imx6ul-mba6ul {
+		pinctrl_lcdif: lcdifgrp {
+			fsl,pins = <
+				MX6UL_PAD_LCD_DATA00__LCDIF_DATA00	0x79
+				MX6UL_PAD_LCD_DATA01__LCDIF_DATA01	0x79
+				MX6UL_PAD_LCD_DATA02__LCDIF_DATA02	0x79
+				MX6UL_PAD_LCD_DATA03__LCDIF_DATA03	0x79
+				MX6UL_PAD_LCD_DATA04__LCDIF_DATA04	0x79
+				MX6UL_PAD_LCD_DATA05__LCDIF_DATA05	0x79
+				MX6UL_PAD_LCD_DATA06__LCDIF_DATA06	0x79
+				MX6UL_PAD_LCD_DATA07__LCDIF_DATA07	0x79
+				MX6UL_PAD_LCD_DATA08__LCDIF_DATA08	0x79
+				MX6UL_PAD_LCD_DATA09__LCDIF_DATA09	0x79
+				MX6UL_PAD_LCD_DATA10__LCDIF_DATA10	0x79
+				MX6UL_PAD_LCD_DATA11__LCDIF_DATA11	0x79
+				MX6UL_PAD_LCD_DATA12__LCDIF_DATA12	0x79
+				MX6UL_PAD_LCD_DATA13__LCDIF_DATA13	0x79
+				MX6UL_PAD_LCD_DATA14__LCDIF_DATA14	0x79
+				MX6UL_PAD_LCD_DATA15__LCDIF_DATA15	0x79
+				MX6UL_PAD_LCD_DATA16__LCDIF_DATA16	0x79
+				MX6UL_PAD_LCD_DATA17__LCDIF_DATA17	0x79
+				MX6UL_PAD_LCD_DATA18__LCDIF_DATA18	0x79
+				MX6UL_PAD_LCD_DATA19__LCDIF_DATA19	0x79
+				MX6UL_PAD_LCD_DATA20__LCDIF_DATA20	0x79
+				MX6UL_PAD_LCD_DATA21__LCDIF_DATA21	0x79
+				MX6UL_PAD_LCD_DATA22__LCDIF_DATA22	0x79
+				MX6UL_PAD_LCD_DATA23__LCDIF_DATA23	0x79
+			>;
+		};
+		pinctrl_lcdifctrl: lcdifctrlgrp {
+			fsl,pins = <
+				MX6UL_PAD_LCD_CLK__LCDIF_CLK		0x79
+				MX6UL_PAD_LCD_ENABLE__LCDIF_ENABLE	0x79
+				MX6UL_PAD_LCD_HSYNC__LCDIF_HSYNC	0x79
+				MX6UL_PAD_LCD_VSYNC__LCDIF_VSYNC	0x79
+			>;
+		};
+	};
+};
+
+&lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcdif>, <&pinctrl_lcdifctrl>;
+	status = "okay";
+};
+
+&pxp {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6ul-mba6ul-lvds.dtsi b/arch/arm/boot/dts/imx6ul-mba6ul-lvds.dtsi
new file mode 100644
index 000000000000..7613ce941ac0
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ul-lvds.dtsi
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	display_lvds: display {
+		bits-per-pixel = <32>; /* 32: RGB888/666 mode */
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: chimei-g104-x1 {
+				clock-frequency = <65000000>;
+				hactive = <1024>;
+				vactive = <768>;
+				hfront-porch = <75>;
+				hback-porch = <10>;
+				hsync-len = <255>; /* FIXME: typ < 256!*/
+				vback-porch = <100>;
+				vfront-porch = <10>;
+				vsync-len = <38>;
+				de-active = <1>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
+&backlight {
+	pwms = <&pwm2 0 62500>;
+	brightness-levels = <0 16 32 64 96 128 160 192 224 255>;
+	default-brightness-level = <5>;
+	status = "okay";
+};
+
+/* for lcd reset != #stkrst */
+/* place r2021, resolder r2022 and iomux following: */
+/* MX6UL_PAD_LCD_RESET__GPIO3_IO04	0x79 */
+&lcdif {
+	display = <&display_lvds>;
+	lcd-supply = <&reg_lcd_pwr>;
+	status = "okay";
+};
+
+&reg_lcd_pwr {
+	status = "okay";
+};
+
+&reg_lvds_transmitter {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct21.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct21.dts
new file mode 100644
index 000000000000..9adfb7fd9687
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct21.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2017 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct21.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct44.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct44.dts
new file mode 100644
index 000000000000..0c884bb890aa
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-dmb-ct44.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct44.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-glyn-etm0700g0edh6.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-glyn-etm0700g0edh6.dts
new file mode 100644
index 000000000000..afe3cc15dd7d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-glyn-etm0700g0edh6.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct21.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct21.dts
new file mode 100644
index 000000000000..fd2d30fc01d7
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct21.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2017 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct21.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct44.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct44.dts
new file mode 100644
index 000000000000..458ebbe6f7a3
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-dmb-ct44.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct44.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-lga-glyn-etm0700g0edh6.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-glyn-etm0700g0edh6.dts
new file mode 100644
index 000000000000..4f2728eb4b5f
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-glyn-etm0700g0edh6.dts
@@ -0,0 +1,15 @@
+
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-lga-lvds.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-lvds.dts
new file mode 100644
index 000000000000..d9c7411d7c66
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-lga-lvds.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-lvds.dtsi"
diff --git a/arch/arm/boot/dts/imx6ul-mba6ulx-lvds.dts b/arch/arm/boot/dts/imx6ul-mba6ulx-lvds.dts
new file mode 100644
index 000000000000..d834fd6381e3
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-mba6ulx-lvds.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016 Marco Felsch, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ul-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-lvds.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct21.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct21.dts
new file mode 100644
index 000000000000..440ba7da6ad6
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct21.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct21.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct44.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct44.dts
new file mode 100644
index 000000000000..a1462faa51af
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-dmb-ct44.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct44.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-glyn-etm0700g0edh6.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-glyn-etm0700g0edh6.dts
new file mode 100644
index 000000000000..c7b738764750
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-glyn-etm0700g0edh6.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct21.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct21.dts
new file mode 100644
index 000000000000..21cd84abb566
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct21.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct21.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct44.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct44.dts
new file mode 100644
index 000000000000..ca5c2c9b9e4b
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-dmb-ct44.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-dmb-ct44.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-lga-glyn-etm0700g0edh6.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-glyn-etm0700g0edh6.dts
new file mode 100644
index 000000000000..4fc9224de041
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-glyn-etm0700g0edh6.dts
@@ -0,0 +1,15 @@
+
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-glyn-etm0700g0edh6.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-lga-lvds.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-lvds.dts
new file mode 100644
index 000000000000..2dfaf55793ee
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-lga-lvds.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx-lga.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-lvds.dtsi"
diff --git a/arch/arm/boot/dts/imx6ull-mba6ulx-lvds.dts b/arch/arm/boot/dts/imx6ull-mba6ulx-lvds.dts
new file mode 100644
index 000000000000..22dae7f1dadf
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-mba6ulx-lvds.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2018 Markus Niebel, TQ Systems GmbH
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6ull-mba6ulx.dts"
+#include "imx6ul-mba6ul-lcdif.dtsi"
+#include "imx6ul-mba6ul-lvds.dtsi"
